analyses/focal-cn-file-preparation/06-find-recurrent-calls.Rmd

```diff
@@ -0,0 +1,114 @@
+date: 2020
+---
+
+This notebook determines the recurrent focal copy number dominant status calls by region using the output of `05-define-most-focal-cn-units.Rmd`.
+
+Recurrence here has been arbitrarily defined based on the plotting of the distribution of status calls and a [similar decision](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/66bb67a7bf29aad4510a0913a2dbc88da0013be8/analyses/fusion_filtering/06-recurrent-fusions-per-histology.R#L152) made in `analyses/fusion_filtering/06-recurrent-fusions-per-histology.R` to make the cutoff for recurrence to be greater than a count of 3 samples that have the same CN status call in the same region.
+
+## Usage
+
+This notebook is intended to be run from the command line with the following (assumes you are in the root directory of the repository):
+
+```
+Rscript -e "rmarkdown::render('analyses/focal-cn-file-preparation/06-find-recurrent-calls.Rmd', clean = TRUE)"
+```
+
+## Set up
+
+### Cutoffs
+
+```{r}
+# Here, we arbitrarily define recurrence to be greater than 3 based on 
+# the plot above and on a similar decision made in
+# `analyses/fusion_filtering/06-recurrent-fusions-per-histology.R`
+# https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/66bb67a7bf29aad4510a0913a2dbc88da0013be8/analyses/fusion_filtering/06-recurrent-fusions-per-histology.R#L152
+recurrence_cutoff <- 3
+```
+
+
+### Libraries and functions
+
+```{r}
+library(tidyverse)
+```
+
+### Files and directories
+
+```{r}
+results_dir <- "results"
+```
+
+### Read in files
+
+Read in focal CN dominant status calls prepared in `05-define-most-focal-cn-units.Rmd`.
+
+```{r}
+focal_cn_df <- read_tsv(file.path(results_dir, "consensus_seg_most_focal_cn_status.tsv.gz"))
+```
+
+## Find recurrent calls
+
+### Tally the sample counts of each unique status call by region.
+
+```{r}
+# Tally the number of samples that have the same status call by region
+final_df_tally <- focal_cn_df %>%
+  group_by(status, region, region_type) %>%
+  tally(name = "count")
+
+# Display table with counts of each unique status call by region
+final_df_tally %>%
+  arrange(desc(count))
+```
+
+### Plot the distribution of counts
+
+```{r}
+# Plot the distribution of counts
+ggplot(final_df_tally, aes(x = count)) +
+  geom_histogram(bins = 100) +
+  geom_vline(aes(xintercept = recurrence_cutoff), linetype = "longdash", color = "green", size = .4)
+```
+
+The `recurrence_cutoff` defined as 3 appears to be reasonable as seen in the distribution of counts plot above.
+
+### Isolate recurrent calls
+
+```{r}
+# Create a data frame of only observations that have counts greater than the `recurrence_cutoff`
+recurrent_df <- final_df_tally %>%
+  filter(count > recurrence_cutoff)
+
+# Display table
+recurrent_df
+```
+
+### Filter focal CN data frame to include only recurrent observations
+
+```{r}
+# Filter the focal CN data frame to include only the recurrent observations
+final_df_recurrent <- focal_cn_df %>%
+  select(-region_type) %>%
```

Comment:
Why are we getting rid of region_type here?

Reply:
I did this to avoid the duplicate column that would come from the `inner_join` step below (there's probably a better way to do this here).

Reply:
This is fine to me. Maybe just add a comment to explain why.

Reply:
I _think_ including `region_type` in your `by` below would accomplish the same thing and potentially be a little friendlier to your reader
```R
   inner_join(recurrent_df, by = c("region", "status",  "region_type"))
```

Reply:
Ah, I'll give that a try @jaclyn-taroni, thank you!